src/Symfony/Component/Config/Definition/Builder/TreeBuilder.php

```diff
@@ -43,9 +44,13 @@ public function __construct(string $name = null, string $type = 'array', NodeBui
      * @return ArrayNodeDefinition|NodeDefinition The root node (as an ArrayNodeDefinition when the type is 'array')
      *
      * @throws \RuntimeException When the node type is not supported
+     *
+     * @deprecated since Symfony 4.3, pass the root name to the constructor instead
      */
     public function root($name, $type = 'array', NodeBuilder $builder = null)
     {
+        @trigger_error(sprintf('The "%s()" method called for the "%s" configuration is deprecated since Symfony 4.3, pass the root name to the constructor instead.', __METHOD__, $name), E_USER_DEPRECATED);
```

Comment:
idea: we may want to tweak the deprecation message of the constructor in the `4.2` in the same way to mention the configuration

Reply:
When the constructor is called without the root name, we can not mention the not given name in the deprecation message ;)

Reply:
🙈